examples/di/InfiAgent-DABench/DABench.py

```diff
@@ -0,0 +1,57 @@
+import json
+from pathlib import Path
+from metagpt.const import DABENCH_PATH
+from examples.di.requirements_prompt import DABENCH
+class DABench:
+    def __init__(self, questions_file=Path(DABENCH_PATH) / 'da-dev-questions.jsonl', answers_file=Path(DABENCH_PATH) / 'da-dev-labels.jsonl', template = ''):
+        # Read questions from a JSONL file
+        with open(questions_file, 'r') as file:
+            self.questions = {int(json.loads(line)['id']): json.loads(line) for line in file}
+
+        # Read answers from a JSONL file
+        with open(answers_file, 'r') as file:
+            self.answers = {int(json.loads(line)['id']): json.loads(line) for line in file}
+
+        self.template = template if template else DABENCH
+    def get_question(self, question_id):
+        """Retrieve the question by its id."""
+        return self.questions.get(question_id, "Question not found.")
+
+    def get_prompt(self, question_id):
+        """Retrieve the question by its id."""
+        temp = self.get_question(question_id)
+        return self.template.format(question=temp['question'], constraints=temp['constraints'], format=temp['format'], file_name= str(DABENCH_PATH) + '/da-dev-tables/' + temp['file_name'], level=temp['level'],)
+
+    def get_answer(self, answer_id):
+        """Retrieve the answer list by its id."""
+        return self.answers.get(answer_id, "Answer not found.")
+
+    def eval(self, id, prediction):
```

Comment:
Should add the reference : https://github.com/InfiAgent/InfiAgent/blob/main/examples/DA-Agent/eval_closed_form.py

Reply:
ok